CompilerFolderHandling/New-BcCompilerFolder.ps1

```diff
@@ -51,12 +51,12 @@ try {
     $version = [System.Version]($parts[4])
     $country = $parts[5]
 
-    $vsixFile = DetermineVsixFile -vsixFile $vsixFile
-
     if ($version -lt "16.0.0.0") {
         throw "Containerless compiling is not supported with versions before 16.0"
     }
     
+	$vsixFile = GetAlLanguageExtension -vsixFile $vsixFile -extract;

```

Comment:
If running on a GitHub hosted runner (i.e. no local cache), you will be downloading and extracting the AL Language extention with every compiler folder and basically disabling the GitHub cache.
The cacheFolder might already be populated with compiler, apps and everything when we get here.

Maybe you could move this to line 81 - and only do this if $compilerPath doesn't exist ?